Implements a way of branching behavior without if/else

hugobessa opened the pull request:
Creates a way to avoid nesting code with if/else when changing logic whether flag/switch is active or not. This force the users of the lib of creating more readable and modular code.

It also makes easier to implement automatic removal of flags in the code base using Abstract Syntax Tree tools like astroid.

laisvarejao commented on waffle/tests/test_decorators.py:
```diff
@@ -102,3 +104,13 @@ def test_flag_must_be_inactive_and_not_redirect(self):
         resp = self.client.get('/flagged_view_with_invalid_redirect')
         self.assertEqual(200, resp.status_code)
+
+    def test_waffle_callable_decorated_function_only_called_if_called_twice(self):
```
Why should it only be called if called twice?

hugobessa replied:
Because in the first time you call you create a function that, when run with no parameters, will execute with the parameters. The second time, you call this function

laisvarejao commented on waffle/helpers.py:
```diff
@@ -0,0 +1,62 @@
+        'required but if present it has to be a `WaffleCallable`'
+    )
+
+
```
2 spaces

laisvarejao reviewed (approved):
LGTM! Just commented a really small fix, but feel free to merge after that!

hugobessa committed:
Implements a way of branching behavior without if/else

Creates a way to avoid nesting code with if/else when changing logic
wether flag is active or not. This force the users of the lib of
creating more readable and modular code.

It also makes easier to implement automatic removal of flags in the
code base using Abstract Syntax Tree tools like astroid.
